src/wp-includes/fonts/class-wp-font-family.php

```diff
@@ -0,0 +1,615 @@
+	 * @param string $filename The filename to be used for the uploaded file.
+	 * @return array The overrides for the 'wp_handle_upload' function.
+	 */
+	private function get_upload_overrides( $filename ) {
+		return array(
+			// Arbitrary string to avoid the is_uploaded_file() check applied
+			// when using 'wp_handle_upload'.
+			'action'                   => 'wp_handle_font_upload',
+			// Not testing a form submission.
+			'test_form'                => false,
+			// Seems mime type for files that are not images cannot be tested.
+			// See wp_check_filetype_and_ext().
+			'test_type'                => true,
+			'mimes'                    => WP_Font_Library::get_expected_font_mime_types_per_php_version(),
+			'unique_filename_callback' => static function () use ( $filename ) {
+				// Keep the original filename.
+				return $filename;
+			},
+		);
+	}
+
+	/**
+	 * Downloads a font asset from a specified source URL and saves it to
+	 * the font directory.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $url      The source URL of the font asset to be downloaded.
+	 * @param string $filename The filename to save the downloaded font asset as.
+	 * @return string|bool The relative path to the downloaded font asset.
+	 *                     False if the download failed.
+	 */
+	private function download_asset( $url, $filename ) {
+		// Checks if the file to be downloaded has a font mime type.
+		if ( ! WP_Font_Family_Utils::has_font_mime_type( $filename ) ) {
+			return false;
+		}
+
+		// Include file with download_url() if function doesn't exist.
+		if ( ! function_exists( 'download_url' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		// Downloads the font asset or returns false.
+		$temp_file = download_url( $url );
+		if ( is_wp_error( $temp_file ) ) {
+			return false;
+		}
+
+		$overrides = $this->get_upload_overrides( $filename );
+
+		$file = array(
+			'tmp_name' => $temp_file,
+			'name'     => $filename,
+		);
+
+		$handled_file = wp_handle_upload( $file, $overrides );
+
+		// Cleans the temp file.
+		@unlink( $temp_file );
+
+		if ( ! isset( $handled_file['url'] ) ) {
+			return false;
+		}
+
+		// Returns the relative path to the downloaded font asset to be used as
+		// font face src.
+		return $handled_file['url'];
+	}
+
+	/**
+	 * Moves an uploaded font face asset from temp folder to the fonts directory.
+	 *
+	 * This is used when uploading local fonts.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $font_face Font face to download.
+	 * @param array $file      Uploaded file.
+	 * @return array New font face with all assets downloaded and referenced in
+	 *               the font face definition.
+	 */
+	private function move_font_face_asset( $font_face, $file ) {
+		$new_font_face = $font_face;
+		$filename      = WP_Font_Family_Utils::get_filename_from_font_face(
+			$this->data['slug'],
+			$font_face,
+			$file['name']
+		);
+
+		// Remove the uploaded font asset reference from the font face definition
```

Comment:
Looks like the PR needs to be updated to use the multi-line comment format in a number of areas. I'll stop commenting on each case 😝

Reply:
Thanks @costdev. I was updating each of those instances earlier and got sidetracked. Each will be updated before it's committed. Thank you!

Reply:
@costdev I've updated all of the multi-line comments in commit https://github.com/WordPress/wordpress-develop/pull/5285/commits/e948c1b764370bba2fa982b190e3c9dbb179cfab ✅